### PR TITLE
Angular App

### PR DESCRIPTION
**Pull request by dat26061994**
To day i do:
- Create template Member Page
- using Angular Js to CRUD member Page

### CODE REVIEW

File: app/Http/Middleware/RedirectIfAuthenticated.php [hunk 1]
```diff
@@ -10,17 +10,28 @@ class RedirectIfAuthenticated
...
+     * @param  \Closure $next
+     * @param  string|null $guard
      * @return mixed
      */
     public function handle($request, Closure $next, $guard = null)
     {
-        if (Auth::guard($guard)->check()) {
-            return redirect('/home');
+
+        switch ($guard) {
```
Comment (kakeyang) [inline app/Http/Middleware/RedirectIfAuthenticated.php:21]: Why did you use switch statement here? If statement seems to be enough.

File: app/Http/Middleware/RedirectIfAuthenticated.php [hunk 2]
```diff
@@ -10,17 +10,28 @@ class RedirectIfAuthenticated
     /**
      * Handle an incoming request.
      *
-     * @param  \Illuminate\Http\Request  $request
-     * @param  \Closure  $next
-     * @param  string|null  $guard
+     * @param  \Illuminate\Http\Request $request
```
Comment (4c0n) [inline app/Http/Middleware/RedirectIfAuthenticated.php:13]: You can add a type hint for Request in the method definition as well.

File: public/admin/js/mem.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            case 'edit':
+                $scope.frmTitle = 'Edit Member';
+                $http({
+                    method: 'GET',
+                    url: API + 'edit/' + id
+                }).then(function successCallback(response) {
+                    $scope.id = id;
+                    $scope.member = response.data;
+                }, function errorCallback() {
+                    alert('ERROR');
```
Comment (kakeyang) [inline public/admin/js/mem.js:57]: I don't understand what this process does but this string is enough for users to understand?

File: app/Http/Middleware/RedirectIfAuthenticated.php [hunk 4]
```diff
@@ -10,17 +10,28 @@ class RedirectIfAuthenticated
...
      * @return mixed
      */
     public function handle($request, Closure $next, $guard = null)
     {
-        if (Auth::guard($guard)->check()) {
-            return redirect('/home');
+
+        switch ($guard) {
+            case 'admin':
+                if (Auth::guard($guard)->chech()) {
```
Comment (4c0n) [inline app/Http/Middleware/RedirectIfAuthenticated.php:23]: chech()?

File: app/Member.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Member extends Model
```
Comment (4c0n) [inline app/Member.php:7]: Where are the doc blocks in this class?

File: public/admin/js/mem.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
+var app = angular.module('my-app', ['angularUtils.directives.dirPagination']).constant('API', 'http://localhost:8080/demoApp2/admin/');
```
Comment (4c0n) [inline public/admin/js/mem.js:1]: Does this need to contain the full URL?

File: resources/assets/js/mem.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+/**
```
Comment (4c0n) [inline resources/assets/js/mem.js:1]: This file only contains this comment?

**Review by 4c0n — CHANGES_REQUESTED** [review]

**Pull request merged**